tutorials/ios-chatbot/index.md

```diff
@@ -0,0 +1,145 @@
+
+
+### Create Agent
+
+The first step is to create a "Tour Guide Agent." 
+
+
+### Create Intents
+
+![alt_text](chatbots-3.png "Create Intents Screenshot")
+
+To engage users in a conversation, we first need to understand what users are saying to the agent and we do that with intents and entities. Intents map what your users say to what your conversational experience should do. Entities are used to extract parameter values from use queries.
+
+Each intent contains a set of examples of user input and the desired automated response. To do that, you need to predict what users will say to open the conversation, and then enter those phrases in the "Add user expression" box. This list doesn't need to be comprehensive. API.AI uses Machine Learning to train the agent to understand more variations of these examples. Later on, you can train the API.AI agent to understand more variations. For example, go to the Default Welcome Intent and add some user expressions "how are you", "hello", "hi" to open the conversation.
+
+The next step after that is to add some more text responses. 
+
+![Default Welcome Intent Screenshot](chatbots-6.png "Default Welcome Intent Screenshot")
+
+
+Next, it's time to work on contexts.
+
+
+### Contexts
+
+_[Contexts](https://docs.api.ai/docs/concept-contexts)_ represent the current context of a user's request. They are helpful for differentiating phrases that may be vague or have different meanings depending on the user's preferences or geographic location, the current page in an app, or the topic of conversation. Let's look at an example.
+
+_User: Where am I?_
+
+_Bot: Please upload a nearby picture and I can help find out where you are._
+
+_[User uploads a picture of Golden Gate Bridge.]_
+
+_Bot: You are near Golden Gate Bridge._
+
+_User: How much is the ticket?_
+
+_Bot: Golden Gate Bridge is free to visit._
+
+_User: When does it close today?_
+
+_Bot: Golden Gate Bridge is open 24 hours a day, 7 days a week._
+
+_User: How do I get there?_
+
+_[Bot shows a map to Golden Gate Bridge.]_
+
+In the above conversation, when user asks "How much is the ticket?" and "When does it close today?" or "How do I get there?", the bot understands that the context is around Golden Gate Bridge.
+
+The next thing to do is to weave intents and contexts together. For our example, each box in the diagram below is an intent and a context; the arrows indicate the relationships between them.
+
+![alt_text](chatbots-7.png "Contexts Relationship")
+
+
+
+### Output Contexts
+
+Contexts are tied to user sessions (a session ID that you pass in API calls). If a user expression is matched to an intent, the intent can then set an output context to be shared by this expression in the future. You can also add a context when you send the user request to your API.AI agent. In our example, the where intent sets the where output context so that Location intent will be matched in the future.
```

Comment:
Suggestion for the last sentence:
In our example, the **Where** intent sets the **Where** output context so that **Location** intent will be matched in the future.